janus/lib/client/jsx/flags/flags-context.tsx

```diff
@@ -0,0 +1,52 @@
+import React, {useState, useEffect, createContext} from 'react';
+
+import {UserFlagsInterface, Project} from '../types/janus_types';
+
+const defaultFlagsState = {
+  users: [] as UserFlagsInterface[],
+  projects: [] as Project[]
+};
+
+export type FlagsState = Readonly<typeof defaultFlagsState>;
+
+export const defaultContext = {
+  state: defaultFlagsState as FlagsState,
+  setUsers: (users: UserFlagsInterface[]) => {},
+  setProjects: (projects: Project[]) => {}
+};
+
+export type FlagsContextData = typeof defaultContext;
+export const FlagsContext = createContext(defaultContext);
+export type FlagsContext = typeof FlagsContext;
+export type ProviderProps = {
+  params?: {};
+  children: any;
+};
+
+export const FlagsProvider = (
```

Comment:
@corps , when we reviewed the Vulcan UI changes you made, you mentioned not using Redux with a Context. Is this approach more like what you  meant? This way we avoid using `dispatch` and the async nature of it, just keep the `state` via `useState` in the Context?

Reply:
Yeah, i like this approach alot.  You can even choose to expose 'action' functions.  So instead of putting setState or setProjects into the context, you could `useCallback` and create little functions that modify the state like an action, and pass those to the context only if you preferred.  Many options for organizing.  This works great.